### DISCUSSION TITLE
The server that the application is connecting to is unverified (Android)

### DISCUSSION THREAD

**Discussion by kazrazi** [airsdk/Adobe-Runtime-Support#2683] [General]
The server that the application is connecting to is unverified (Android).
Our applications receive such a warning at startup. The user can click on "trust for this session", but this is very bad for the players. Is it possible to fix this? Our system administrator said that we need to add a root certificate to the application, but is it possible to do this in AIR?

**Comment by ajwfrost** [comment]
How are you making the connection? `SecureSocket` does have a way for you to provide a trusted certificate that could then be seen as the root certificate to validate your server cert:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/SecureSocket.html#addBinaryChainBuildingCertificate()

If it's a self-signed certificate then we're also just implementing a mechanism to allow the AS3 code to have a first chance to decide whether to continue or cancel the connection, see #1453 
To start with this will be provided as an interim bit of functionality without an AS3 API change, but then AIR 51 will include the language updates to simplify this.

thanks

**Reply by kazrazi** [reply]
Thanks, this looks like what I need
